engine/setup/src/c/setup.cpp

```diff
@@ -87,11 +87,31 @@ static void changeToProgramDirectory( char *argv0 )
 typedef char FileNameCharType[65536];
 int WINAPI WinMain( HINSTANCE hInst, HINSTANCE hPrevInst, LPSTR lpCmdLine, int nShowCmd )
 {
-    char  *argv0 = new char[65535];
-    char **argv  = &argv0;
-    int    argc  = 0;
-    strcpy( argv0, origpath );
-    GetModuleFileName( NULL, argv0, 65534 );
+    char **argv;
+    int  argc;
+    //strcpy( argv0, origpath );
+    //GetModuleFileName( NULL, argv0, 65534 );
+    
+    LPWSTR* szArglist;
+    int i;
+    argv = (char**)malloc(3*sizeof(char*));
+    szArglist = CommandLineToArgvW(GetCommandLineW(), &argc);
```

Comment:
As these are wide-string (multi-byte) functions; then it should be sized based on `wchar_t` instead of `char`, same below. Otherwise the ASCII version of GetCommandLineW() should be used - GetCommandLineA().

It would certainly be preferable to stick to the wide-string version if we can; one less thing to do later if we want to support multiple languages all the way through.

Reply:
Good catch - we should probably use the ascii version for the moment because I think the rest of setup is still assuming char*  in other places -- but would be best to shift everything to wide-char in the not-too-distant future.

Reply:
Simplified that whole block to be ASCII (for the moment) and much shorter - there are actually intrinsics for __argc and __argv available, so the windows side just has to declare argv and argc and set them to the intrinsics.